### PR TITLE
fix: do not load kb for project [bugfix] (BUG-1172)

### PR DESCRIPTION
**Pull request by DecathectZero**
Testing across the following areas.

Works for KB base preview 
Screenshot 2024-03-07 at 5 24 57 PM

Works for KB preview in step
Screenshot 2024-03-07 at 5 27 15 PM

Works for prototype KB step
Screenshot 2024-03-07 at 5 28 00 PM

Works for KB Query API
Screenshot 2024-03-07 at 5 31 56 PM

KB No Match Fallback With KB
Screenshot 2024-03-07 at 5 31 56 PM
KB No Match Fallback Without KB
Screenshot 2024-03-07 at 5 54 17 PM
Screenshot 2024-03-07 at 5 54 26 PM

Fetching Version Publishing
Screenshot 2024-03-07 at 5 55 58 PM

### CODE REVIEW

File: lib/services/aiSynthesis/index.ts [hunk 1]
```diff
@@ -286,14 +286,21 @@ class AISynthesis extends AbstractManager {
...
+
         runtime.trace.addTrace({
           type: 'knowledgeBase',
           payload: {
-            chunks: data.chunks.map(({ score, documentID }) => ({
-              score,
-              documentID,
-              documentData: runtime.project?.knowledgeBase?.documents[documentID]?.data,
-            })),
+            chunks: await Promise.all(
```
Comment (trs) [inline lib/services/aiSynthesis/index.ts:297]: Don't need the promise all and async mapping, since you're loading documents above

Comment (DecathectZero) [reply lib/services/aiSynthesis/index.ts:297]: sorry this was some leftover code from another solution I was trying, was in too much of a rush

Comment (DecathectZero) [reply lib/services/aiSynthesis/index.ts:297]: great catch

File: lib/services/aiSynthesis/index.ts [hunk 2]
```diff
@@ -286,13 +286,18 @@ class AISynthesis extends AbstractManager {
...
+        const documents = await runtime.getKBDocuments(
+          projectID,
+          data.chunks.map((chunk) => chunk.documentID)
+        );
+
         runtime.trace.addTrace({
           type: 'knowledgeBase',
           payload: {
-            chunks: data.chunks.map(({ score, documentID }) => ({
+            chunks: data.chunks.map(async ({ score, documentID }) => ({
```
Comment (trs) [inline lib/services/aiSynthesis/index.ts:297]:
```suggestion
            chunks: data.chunks.map(({ score, documentID }) => ({
```

File: runtime/lib/DataAPI/mongoDataAPI.ts [hunk 3]
```diff
@@ -71,16 +70,59 @@ class MongoDataAPI<
...
+      {
+        $project: {
+          hasDocuments: {
+            $gt: [{ $size: { $objectToArray: '$knowledgeBase.documents' } }, 0],
+          },
+        },
+      },
+    ]);
+
+    return result.hasNext();
```
Comment (trs) [inline runtime/lib/DataAPI/mongoDataAPI.ts:106]: Does this work? Or do you need to check the projected `hasDocuments` from the result?

`result?.[0]?.hasDocuments`

Comment (DecathectZero) [reply runtime/lib/DataAPI/mongoDataAPI.ts:106]: I need to test this still

Comment (DecathectZero) [reply runtime/lib/DataAPI/mongoDataAPI.ts:106]: Yup I checked and it works.
Screenshot 2024-03-07 at 5 46 35 PM

Comment (DecathectZero) [reply runtime/lib/DataAPI/mongoDataAPI.ts:106]: Nope, you are right, don't trust copilot on everything.

**Review by trs — APPROVED** [review]

**Pull request closed without merging**